src/xia2/Modules/MultiCrystal/data_manager.py

```diff
@@ -27,22 +27,44 @@ def __init__(self, experiments, reflections):
             value: key for key, value in self.ids_to_identifiers_map.items()
         }
 
+        if all(e.scan is None for e in self._experiments):
+            self.all_stills = True
+        elif all(e.scan is not None for e in self._experiments):
+            self.all_stills = False
+        else:
+            raise ValueError(
+                "cannot mix stills and rotation data for multi crystal analysis"
+            )
+
         self._set_batches()
 
     def _set_batches(self):
-        max_batches = max(e.scan.get_image_range()[1] for e in self._experiments)
-        max_batches += 10  # allow some head room
+
+        if not self.all_stills:
+            max_batches = max(e.scan.get_image_range()[1] for e in self._experiments)
+            max_batches += 10  # allow some head room
+        else:
+            max_batches = 1
+            self.ssx_batch_dict = {}
+            self.ssx_batch_list = []
 
         n = int(math.ceil(math.log10(max_batches)))
 
         for i, expt in enumerate(self._experiments):
-            expt.scan.set_batch_offset(i * 10**n)
-            if expt.imageset:
-                # This may be a different scan instance ¯\_(ツ)_/¯
-                expt.imageset.get_scan().set_batch_offset(expt.scan.get_batch_offset())
-            logger.debug(
-                f"{expt.scan.get_batch_offset()} {expt.scan.get_batch_range()}"
-            )
+            if not self.all_stills:
+                expt.scan.set_batch_offset(i * 10**n)
+                if expt.imageset:
+                    # This may be a different scan instance ¯\_(ツ)_/¯
+                    expt.imageset.get_scan().set_batch_offset(
+                        expt.scan.get_batch_offset()
+                    )
+                logger.debug(
+                    f"{expt.scan.get_batch_offset()} {expt.scan.get_batch_range()}"
+                )
+
+            else:
+                self.ssx_batch_dict[expt] = i * 10**n
```

Comment:
I don't think the ssx_batch_dict is used for anything? So we can remove it.

Reply:
Removed :)